docs/tools/node_maintain.md

```diff
@@ -0,0 +1,84 @@
+# node_maintain
+
+## Overview
+
+Graceful decommissioning could minimize the impact on existing job when remove unhealthy nodes. 
+This is a tool to help you achieve it in OpenPAI. 
+
+In OpenPAI, an unhealthy node-list records the nodes to be decommissioned. 
+Components could sync the list and decommission them as needed.
+
+**NOTES: For now, only YARN nodes are decommissioned, HDFS doesn't.**
+
+## Commands
+
+Currently, node-list update and component refresh are both manual actions. 
+This tool provides corresponding command for them. 
+The only mandatory argument for all commands is `master_ip`. Usually, components share the same master ip,
+in different case(i.e. load-balance), you could overwrite `master_ip` separately by `--kubernetes-ip` and `--yarn-ip`
+
+So, a common invoking looks like:
+```bash
+python node_maintain.py {master_ip} sub-command sub-command-arguments
```

Comment:
>node_maintain.p [](start = 7, length = 15)

does the command has to be run in the path /src/tools? If so, please state it in the manual.